detoks/src/main/java/nl/tudelft/trustchain/detoks/gossiper/TorrentGossiper.kt

```diff
@@ -32,27 +32,34 @@ class TorrentGossiper(
         val deToksCommunity = IPv8Android.getInstance().getOverlay<DeToksCommunity>()!!
         val randomPeers = pickRandomN(deToksCommunity.getPeers(), peers)
         val handlers = TorrentManager.getInstance(context).getListOfTorrents()
+        val profile = TorrentManager.getInstance(context).profile
         if (randomPeers.isEmpty() || handlers.isEmpty()) return
 
         val max = if (handlers.size < blocks) handlers.size else blocks
         val randomMagnets = handlers.random(max).map { it.makeMagnetUri() }
 
         if(randomMagnets.isNotEmpty())
             randomPeers.forEach {
+                val data = randomMagnets
+                    .map { it2 ->
+                        val key = MagnetLink.hashFromMagnet(it2)
+                        profile.incrementHopCount(key)
+                        Pair(it2, profile.profiles[MagnetLink.hashFromMagnet(it2)]!!.hopCount)
```

Comment:
I think this is actually not correct, since now we just increase the hopcount everytime we send it. The purpose of the field was to keep track how many hops were done from the source, so it should only be incremented at the destination. So either send hopcount + 1 without assigning it at this node (probably the right way), or increment it when receiving. 

I would just do addProfile instead for the check if a profile exists or not, and then add + 1 in the next line.
If you agree don't forget to remove `incrementHopCount()`
```suggestion
                        profile.addProfile(key)
                        Pair(it2, profile.profiles[MagnetLink.hashFromMagnet(it2)]!!.hopCount + 1)
```